src/otx/engine/engine.py

```diff
@@ -0,0 +1,407 @@
+            logger (Logger | Iterable[Logger] | bool | None, optional): The logger(s) to be used. Defaults to None.
+            resume (bool, optional): If True, tries to resume training from existing checkpoint.
+            **kwargs: Additional keyword arguments for pl.Trainer configuration.
+
+        Returns:
+            dict[str, Any]: A dictionary containing the callback metrics from the trainer.
+
+        Example:
+            >>> engine.train(
+                max_epochs=3,
+                seed=1234,
+                deterministic=False,
+                precision="32",
+            )
+
+        CLI Usage:
+            1. you can train with data_root only. then OTX will provide default model.
+                ```python
+                otx train --data_root <DATASET_PATH>
+                ```
+            2. you can pick a model or datamodule as Config file or Class.
+                ```python
+                otx train
+                --data_root <DATASET_PATH>
+                --model <CONFIG | CLASS_PATH_OR_NAME> --data <CONFIG | CLASS_PATH_OR_NAME>
+                ```
+            3. Of course, you can override the various values with commands.
+                ```python
+                otx train
+                    --data_root <DATASET_PATH>
+                    --max_epochs <EPOCHS, int> --checkpoint <CKPT_PATH, str>
+                ```
+            4. If you have a complete configuration file, run it like this.
+                ```python
+                otx train --data_root <DATASET_PATH> --config <CONFIG_PATH, str>
+                ```
+        """
+        lit_module = self._build_lightning_module(
+            model=self.model,
+            optimizer=self.optimizer,
+            scheduler=self.scheduler,
+        )
+        lit_module.meta_info = self.datamodule.meta_info
+
+        if seed is not None:
+            seed_everything(seed, workers=True)
+
+        self._build_trainer(
+            logger=logger,
+            callbacks=callbacks,
+            precision=precision,
+            max_epochs=max_epochs,
+            deterministic=deterministic,
+            val_check_interval=val_check_interval,
+            **kwargs,
+        )
+        fit_kwargs: dict[str, Any] = {}
+        if resume:
+            fit_kwargs["ckpt_path"] = self.checkpoint
+        elif self.checkpoint is not None:
+            loaded_checkpoint = torch.load(self.checkpoint)
+            lit_module.load_state_dict(loaded_checkpoint["state_dict"])
+
+        self.trainer.fit(
+            model=lit_module,
+            datamodule=self.datamodule,
+            **fit_kwargs,
+        )
+        self.checkpoint = self.trainer.checkpoint_callback.best_model_path
+
+        return self.trainer.callback_metrics
+
+    def test(
+        self,
+        checkpoint: str | Path | None = None,
+        datamodule: EVAL_DATALOADERS | OTXDataModule | None = None,
```

Comment:
In `Engine.test`, we use a combination of `dataloaders` and `datamodule` from `Trainer.test`. The reason i used `datamodule` as a word in the argument is because we use the same `datamodule` in the argument of the Engine constructor. 

for someone who doesn't know Trainer as well as I do, it's confusing to have dataloaders and datamodules in one place, and they're even used almost interchangeably inside `Trainer.test`. So I've combined them.